components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/edit/EditToolbar.kt

```diff
@@ -57,36 +58,41 @@ class EditToolbar(
         }
 
         setOnTextChangeListener { text, _ ->
+            updateClearVisibility()
             editListener?.onTextChanged(text)
         }
     }
 
     private val defaultColor = ContextCompat.getColor(context, R.color.photonWhite)
 
-    internal var cancelViewColor = defaultColor
+    internal var clearViewColor = defaultColor
         set(value) {
             field = value
-            cancelView.setColorFilter(value)
+            clearView.setColorFilter(value)
         }
 
-    private val cancelView = ImageView(context).apply {
+    private fun updateClearVisibility() {
```

Comment:
nit: passing the text from the listener to the function will make writing a test easier since you don't have to bother about mocking the `urlView` class as well.

Optionally, since this is a one-liner, you can inline it into the listener as well.